### PR TITLE
URL is future-proofed to be ready to accept a user search or somethin…

### PR DESCRIPTION
**Pull request by CodyRoy**
…g. String it dynamically built now :^)

### CODE REVIEW

File: app/src/main/java/com/example/cr554/newsapp/MainActivity.java [hunk 1]
```diff
@@ -47,7 +48,8 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     private void fetchNews(){
-        String url = "http://content.guardianapis.com/search?show-fields=thumbnail&q=sports&api-key=e6678f97-7cd9-4156-88f7-f5f508c29ed3";
+        String url = getNewsUrl("sports");
+        //String url = "http://content.guardianapis.com/search?show-fields=thumbnail&q=sports&api-key=e6678f97-7cd9-4156-88f7-f5f508c29ed3";
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/MainActivity.java:52]: Delete stale code! ;-)

File: app/src/main/java/com/example/cr554/newsapp/MainActivity.java [hunk 2]
```diff
@@ -93,12 +95,12 @@ private void redrawRecycler(){
...
-    private String getNewsUrl(){
+    private String getNewsUrl(String topic){
         String baseUri = getString(R.string.baseuri);
         Uri u = Uri.parse(baseUri).buildUpon()
-                .appendQueryParameter("api_key","e6678f97-7cd9-4156-88f7-f5f508c29ed3")
-                .appendQueryParameter("format", "json")
-                .appendQueryParameter("nojsoncallack","1")
+                .appendQueryParameter("show-fields","thumbnail")
+                .appendQueryParameter("q",topic)
+                .appendQueryParameter("api-key", APIKey.FLICKER_API_KEY)
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/MainActivity.java:103]: API keys are tricky, but I think the general consensus is that you should use them as "build properties". https://medium.com/@cassioso/a-strategy-to-secure-your-api-keys-using-gradle-b9c107272860#.ilte1o4ie would give a pretty good idea of what I mean :)

Comment (stswidwinski) [reply app/src/main/java/com/example/cr554/newsapp/MainActivity.java:103]: :+1: for building Uri using the Uri builder classes!

File: app/src/main/java/com/example/cr554/newsapp/Util/APIKey.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.example.cr554.newsapp.Util;
+
+/**
+ * Created by cr554 on 2/22/2017.
+ */
+
+public class APIKey {
+    public static final String FLICKER_API_KEY ="e6678f97-7cd9-4156-88f7-f5f508c29ed3";
```
Comment (stswidwinski) [inline app/src/main/java/com/example/cr554/newsapp/Util/APIKey.java:8]: as noted above ^^. My tutorial also didn't comply to the guidelines, especially since it was such a small project. It's not a big deal for this one, but it might be a big deal in the future so I just want to make you aware of it ^^

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <resources>
     <string name="app_name">NewsApp</string>
     <string name="noNews">No News Was Found</string>
-    <string name="baseuri">https://content.guardianapis.com/</string>
+    <string name="baseuri">http://content.guardianapis.com/search</string>
```
Comment (stswidwinski) [inline app/src/main/res/values/strings.xml:4]: Well, you don't necessarily need/want to drop the secure in https. Why this change?

Comment (CodyRoy) [reply app/src/main/res/values/strings.xml:4]: iirc: I had initially forgot the /search in the URL and so i just added that - it was still not working, so I doublt checked everything and the only difference in what i had written vs what the API Was telling me the baseURI was was the "s" so i dropped that.

That also didnt work because i was using "api_key" instead of "api-key".  I didnt even think to try and re-add the s.
